dev-docs/source/WritingPerformanceTests.rst

```diff
@@ -0,0 +1,154 @@
+      }
+       
+      void test_slow_performance() {
+         // Put in a unit test that will be slow.
+      }
+   };
+
+Only the presence/absence of the word Performance is used to determine
+if it is a Performance test. As with unit tests, your suite name has to
+match the file name.
+
+You may include ASSERT's in the same way as a unit test to check that
+results are meaningful, but that is unnecessary since these tests are
+for performance, not function. Avoid adding so many assert's that it
+would slow down your test.
+
+Running Performance Tests
+#########################
+
+Performance tests targets are not added by default when running
+cmake/make. To add them as ctest targets, enable them with the flag:
+
+.. code-block:: sh
+
+cmake -DCXXTEST_ADD_PERFORMANCE=TRUE
+
+After re-building, you can then run performance tests with the command:
+
+.. code-block:: sh
+
+ctest [-C Release|Debug] -R Performance
+
+And run regular unit tests, excluding the slow performance ones, with:
+
+.. code-block:: sh
+
+ctest [-C Release|Debug] -E Performance
+
+where the -C option is required for multi-configuration builds like
+Visual Studio & XCode.
+
+The resulting .XML files will contain the detailed timing (of just the
+test portion without setUp/tearDown time).
+
+Note that newly added performance tests will not be registered with
+ctest until cmake has been re-run.
+
+Running tests without ctest
+---------------------------
+
+The tests are still built into every test executable, even if you have
+not set the flag. For example,
+
+.. code-block:: sh
+
+AlgorithmsTest --help-tests
+
+will list all the available tests. If you run
+
+.. code-block:: sh
+
+AlgorithmsTest
+
+alone, it will SKIP the Performance Tests. You have to give the name of
+the specific test suite you want to run, e.g,
+
+.. code-block:: sh
+
+AlgorithmsTest MyAlgorithmPerformanceTest
+
+Best Practice Advice
+####################
+
+-  Performance tests are not System Tests. They should test the code at
+   the same granularity as the unit test suite.
+-  Performance tests are not Unit Tests. There is no need to perform
+   lots of assertions on the test results.
+-  Performance tests should perform enough work such that statistically
+   significant performance differences can be measured.
+-  The performance tests are executed often, so ideally they should
+   typically take 0.2 - 2 seconds to run.
+-  Always perform test set-up outside of the test method. That way your
+   timings will only relate to the target code you wish to measure.
+
+Jobs that monitor performance
+#############################
+
+There is a job in Jenkins (our continuous integration system) that runs
+the performance test suite and generates output that enables us to
+easily monitor timings. The job runs a set of `performance tests on the
+master branch of
+Mantid <http://builds.mantidproject.org/view/Tests/job/performance_tests_master/Master_branch_performance_tests/>`__.
```

Comment:
http://builds.mantidproject.org/job/master_performancetests2/ is probably a more relevant link to go here (obviously the text also needs updated to reflect this is a machine at the ESS).